automem/enrichment/runtime_orchestration.py

```
@@ -0,0 +1,291 @@
+    link_semantic_neighbors_fn: Callable[[Any, str], List[Tuple[str, float]]],
+    enrichment_enable_summaries: bool,
+    generate_summary_fn: Callable[[str, Any], Any],
+    utc_now_fn: Callable[[], str],
+    collection_name: str,
+    unexpected_response_exc: Any,
+    logger: Any,
+) -> bool:
+    """Enrich a memory with relationships, patterns, and entity extraction."""
+    graph = get_memory_graph_fn()
+    if graph is None:
+        raise RuntimeError("FalkorDB unavailable for enrichment")
```

Ruff (0.15.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)